boards/nucleo-l432kc/include/periph_conf.h

```diff
@@ -170,6 +170,29 @@ static const spi_conf_t spi_config[] = {
 #define SPI_NUMOF           ARRAY_SIZE(spi_config)
 /** @} */
 
+/**
+ * @name I2C configuration
+ * @{
+ */
+static const i2c_conf_t i2c_config[] = {
+    {
+        .dev            = I2C1,
+        .speed          = I2C_SPEED_NORMAL,
+        .scl_pin        = GPIO_PIN(PORT_B, 6),
```

Comment:
Looking at the [datasheet](https://www.st.com/content/ccc/resource/technical/document/user_manual/e3/0e/88/05/e8/74/43/a0/DM00231744.pdf/files/DM00231744.pdf/jcr:content/translations/en.DM00231744.pdf) (table 8 pages 20-21 , it seems you inverted SCL and SDA pins. It should PB7 for SCL and PB6 for SDA.

``

Reply:
Switched ports:

```c
static const i2c_conf_t i2c_config[] = {
    {
        .dev            = I2C1,
        .speed          = I2C_SPEED_NORMAL,
        .scl_pin        = GPIO_PIN(PORT_B, 7),
        .sda_pin        = GPIO_PIN(PORT_B, 6),
        .scl_af         = GPIO_AF4,
        .sda_af         = GPIO_AF4,
        .bus            = APB1,
        .rcc_mask       = RCC_APB1ENR1_I2C1EN,
        .irqn           = I2C1_ER_IRQn
    }
};

#define I2C_0_ISR           isr_i2c1_er

#define I2C_NUMOF           ARRAY_SIZE(i2c_config)
```

Have SDA connected to D4 (as according to the documentation it's connected to A4 through SB18) and SCL connected to D5 (as it's connected to A5 through SB16). Still not working with the same parameters for `scl_af`, `sda_af` and `irqn`. As soon as I try to `i2c_acquire` the `I2C_DEV(0)` the function hangs.

Also, taking a look at the [Mbed pinout](https://os.mbed.com/platforms/ST-Nucleo-L432KC/), they also seem to have SDA on PB7 and SCL on PB6.

With the [L031K6 this exact configuration](https://github.com/RIOT-OS/RIOT/pull/12025/files#diff-c9593d755ff8c2ee36fdc4510a4e39c9R138) worked. I'm confused.

Reply:
And that doesn't make sense if one look at the CPU datasheet (where possible pin configuration are listed). So there's a mistake in the board datasheet...